imports/node-app/plugins/discount-codes/mutations/removeDiscountCodeFromCart.js

```diff
@@ -0,0 +1,67 @@
+import SimpleSchema from "simpl-schema";
+import ReactionError from "@reactioncommerce/reaction-error";
+import getCart from "../util/getCart.js";
+
+const inputSchema = new SimpleSchema({
+  cartId: String,
+  discountCodeId: String,
+  shopId: String,
+  token: {
+    type: String,
+    optional: true
+  }
+});
+
+/**
+ * @method removeDiscountCodeFromCart
+ * @summary Applies a discount code to a cart
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} input - an object of all mutation arguments that were sent by the client
+ * @param {Object} input.cartId - Cart to remove discount from
+ * @param {Object} input.discountCodeId - Discount code to remove from cart
+ * @param {String} input.shopId - Shop cart belongs to
+ * @param {String} [input.token] - Cart token, if anonymous
+ * @returns {Promise<Object>} An object with the updated cart with the removed discount
+ */
+export default async function removeDiscountCodeFromCart(context, input) {
+  inputSchema.validate(input);
+
+  const { cartId, discountCodeId, shopId, token } = input;
+  const { collections, userHasPermission } = context;
+  const { Cart } = collections;
+
+  // TODO: figure out the correct permission check here
+  // Should it be `discounts`, or `cart`?
+  // How do we determine this check if the user is the cart owner?
+  if (!userHasPermission(["admin", "owner", "discounts"], shopId)) {
+    throw new ReactionError("access-denied", "Access Denied");
+  }
```

Comment:
Same comment as other file. This check doesn't seem necessary, and the later check can use a new `discounts/apply`. Doesn't seem necessary to have a separate `discounts/remove` role. If you can apply, you should be able to remove, right?

Reply:
I think that makes sense. Will keep `discounts/apply`